exporter/awsemfexporter/metric_declaration.go

```diff
@@ -55,16 +75,38 @@ func (md *MetricDeclaration) Init(logger *zap.Logger) (err error) {
 	}
 	md.Dimensions = validDims
 
+	// Compile metric name selector regexes
 	md.metricRegexList = make([]*regexp.Regexp, len(md.MetricNameSelectors))
 	for i, selector := range md.MetricNameSelectors {
 		md.metricRegexList[i] = regexp.MustCompile(selector)
 	}
+
+	// Initialize label matchers
+	for _, lm := range md.LabelMatchers {
+		if err := lm.Init(); err != nil {
+			return err
+		}
+	}
 	return
 }
 
 // Returns true if the given OTLP Metric's name matches any of the Metric Declaration's
-// metric name selectors.
-func (md *MetricDeclaration) Matches(metric *pdata.Metric) bool {
+// metric name selectors and label matchers.
+func (md *MetricDeclaration) Matches(metric *pdata.Metric, labels map[string]string) bool {
+	// If there are label matchers defined, filter against those first
+	if len(md.LabelMatchers) > 0 {
+		hasLabelMatch := false
+		for _, lm := range md.LabelMatchers {
+			if lm.Matches(labels) {
+				hasLabelMatch = true
+				break
```

Comment:
I forgot the logic for Prometheus here? if we have one label matched, we will generate metrics with all the dimension/label sets in the config?

Reply:
In Prometheus, they have only one label matcher. In our case and in the specs for this CR, we have the additional functionality of matching against multiple label matchers for the same metric declaration to provide more flexibility. As long as they match _at least one_ of the label matchers, the metric declaration will be applied.